react/src/components/atoms/forms/InputCurrency/index.js

```diff
@@ -100,10 +98,10 @@ const Currency = (props) => {
               }
               if ((!hasProperty(props, 'min') || newValue >= props.min) && (!hasProperty(props, 'max') || (newValue <= props.max))) {
                 const { showError, errorMsg } = validNumber(newValue, props.min, props.max);
-                context.updateState({
+                inputEl.value = toCurrency(newValue, countDecimals(props.step));
+                context.updateOwnState({
                   showError,
-                  errorMsg,
-                  value: toCurrency(newValue, countDecimals(props.step))
+                  errorMsg
                 }, () => {
                   if (typeof props.onChange === 'function') {
```

Comment:
can we update this to take advantage of new is dependency?
```suggestion
                  if (props.onChange && is.fn(props.onChange)) {
```